analyze/analyses/dominators/emit.rs

```diff
@@ -0,0 +1,298 @@
+            }
+        }
+
+        for id in &self.items {
+            let start_depth = if *id == items.meta_root() { 0 } else { 1 };
+            recursive_add_rows(
+                &mut table,
+                items,
+                &self.tree,
+                start_depth,
+                &mut row,
+                &opts,
+                *id,
+            );
+        }
+
+        if !self.unreachable_items.is_empty() {
+            let unreachable_items_cnt = self.unreachable_items.len();
+            let unreachable_items_size = self
+                .unreachable_items
+                .iter()
+                .map(|id| &items[*id])
+                .map(|item| item.size())
+                .sum::<u32>();
+            let unreachable_items_size_percent =
+                (f64::from(unreachable_items_size)) / (f64::from(items.size())) * 100.0;
+            table.add_row(vec![
+                unreachable_items_size.to_string(),
+                format!("{:.2}%", unreachable_items_size_percent),
+                format!("[{} Unreachable Items]", unreachable_items_cnt),
+            ]);
+        }
+
+        write!(dest, "{}", &table)?;
+        Ok(())
+    }
+
+    #[cfg(feature = "emit_json")]
+    fn emit_json(&self, items: &ir::Items, dest: &mut io::Write) -> Result<(), traits::Error> {
+        fn recursive_add_children(
+            items: &ir::Items,
+            opts: &opt::Dominators,
+            dominator_tree: &BTreeMap<ir::Id, Vec<ir::Id>>,
+            id: ir::Id,
+            obj: &mut json::Object,
+        ) -> Result<(), traits::Error> {
+            add_json_item(items, id, obj)?;
+
+            if let Some(children) = dominator_tree.get(&id) {
+                let mut children = children.to_vec();
+                children.sort_by(|a, b| items.retained_size(*b).cmp(&items.retained_size(*a)));
+
+                let mut arr = obj.array("children")?;
+                for child in children {
+                    let mut obj = arr.object()?;
+                    recursive_add_children(items, opts, dominator_tree, child, &mut obj)?;
+                }
+            }
+
+            Ok(())
+        }
+
+        // Scoping the borrow of `arr` so we can get another object in the next block
+        let mut obj = json::object(dest)?;
+        {
+            let mut arr = obj.array("items")?;
+            for curr_id in &self.items {
+                let mut item = arr.object()?;
+                recursive_add_children(items, &self.opts, &self.tree, *curr_id, &mut item)?;
+            }
+        }
+
+        if !self.unreachable_items.is_empty() {
+            let mut summary_obj = obj.array("summary")?;
+            let mut unreachable_items_obj = summary_obj.object()?;
+            let unreachable_items_cnt = self.unreachable_items.len();
+            let unreachable_items_size = self
+                .unreachable_items
+                .iter()
+                .map(|id| &items[*id])
+                .map(|item| item.size())
```

Comment:
ditto -- perhaps instead of duplicating the unreachable size and percent computation in `emit_text` and `emit_json`, we could push it up into the `DominatorTree` struct so that we don't have to duplicate the code?